internal/services/redisenterprise/redis_enterprise_database_resource.go

```diff
@@ -296,7 +296,7 @@ func resourceRedisEnterpriseDatabaseCreate(d *pluginsdk.ResourceData, meta inter
 				return fmt.Errorf("retrieving %s: %+v", *clusterId, err)
 			}
 
-			if strings.Contains(strings.ToLower(string(resp.Model.Sku.Name)), "flash") {
+			if resp.Model != nil && strings.Contains(strings.ToLower(string(resp.Model.Sku.Name)), "flash") {
```

Comment:
we should be checking the sku is not nil?

Reply:
the `Sku` field is not a pointer so no need to check it 

https://github.com/hashicorp/terraform-provider-azurerm/blob/5a53f9d8d4b0fb220baf5e54099f203584f72f29/vendor/github.com/hashicorp/go-azure-sdk/resource-manager/redisenterprise/2022-01-01/redisenterprise/model_cluster.go#L15